sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/KafkaBundle.java

```diff
@@ -383,19 +457,88 @@ private ExpectedTopicConfiguration createTopicDescription(TopicConfig c) {
     return new KafkaConsumer<>(consumerProperties, keyDeSerializer, valueDeSerializer);
   }
 
+  /**
+   * Creates and registers a message producer with a String key serializer and the default producer
+   * configuration
+   */
+  public <V> MessageProducer<String, V> createAndRegisterProducer(
+      Class<V> valueClass, ObjectMapper objectMapper, String topicName) {
+    return createAndRegisterProducer(
+        String.class, valueClass, objectMapper, topicName);
+  }
+
+  /**
+   * Creates and registers a message producer with a default producer configuration
+   */
+  @SuppressWarnings("unchecked")
+  public <K, V> MessageProducer<K, V> createAndRegisterProducer(
+      Class<K> keyClass, Class<V> valueClass, ObjectMapper objectMapper, String topicName) {
+
+    ProducerRegistration<K, V> producerRegistration =
+        ProducerRegistration.<K, V>builder()
+            .forTopic(topicName)
+            .withDefaultProducer()
+            .withKeySerializer((Serializer<K>) getSerializer(keyClass, objectMapper))
+            .withValueSerializer((Serializer<V>) getSerializer(valueClass, objectMapper))
+            .build();
+
+    return registerProducer(producerRegistration);
+  }
+
+  /**
+   * Creates and registers a message producer with String key serializer and the given producer
+   * configuration
+   */
+  @SuppressWarnings("unchecked")
+  public <V> MessageProducer<String, V> createAndRegisterProducer(
+      Class<V> valueClass, ObjectMapper objectMapper, String topicName, String producerConfigKey) {
+
+    ProducerRegistration<String, V> producerRegistration =
+        ProducerRegistration.<String, V>builder()
+            .forTopic(topicName)
+            .withProducerConfig(producerConfigKey)
+            .withKeySerializer((Serializer<String>) getSerializer(String.class, objectMapper))
+            .withValueSerializer((Serializer<V>) getSerializer(valueClass, objectMapper))
+            .build();
+
+    return registerProducer(producerRegistration);
+  }
+
+  /**
+   * Creates and registers a message producer with the given serializers and producer configuration
+   */
+  @SuppressWarnings("unchecked")
+  public <K, V> MessageProducer<K, V> createAndRegisterProducer(
```

Comment:
The documentation is a bit short. Javadoc will complain that neither the properties nor the return type are documented.